src/meta/src/controller/catalog/mod.rs

```diff
@@ -429,11 +429,37 @@ impl CatalogController {
             .all(&txn)
             .await?;
 
-        let changed = Self::clean_dirty_sink_downstreams(&txn).await?;
+        let updated_table_ids = Self::clean_dirty_sink_downstreams(&txn).await?;
+        let updated_table_objs = if !updated_table_ids.is_empty() {
+            Table::find()
+                .find_also_related(Object)
+                .filter(table::Column::TableId.is_in(updated_table_ids))
+                .all(&txn)
+                .await?
+        } else {
+            vec![]
+        };
 
         if dirty_job_objs.is_empty() {
-            if changed {
+            if !updated_table_objs.is_empty() {
                 txn.commit().await?;
+
+                // Notify the frontend to update the table info.
+                self.notify_frontend(
```

Comment:
When the `incoming_sink` of the table was fixed, the table's catalog was not notified to the frontend.